backend/src/bundles/users/user.repository.ts

```diff
@@ -35,13 +35,45 @@ class UserRepository implements Repository {
             weight: userDetails.weight,
             height: userDetails.height,
             gender: userDetails.gender,
+            referralCode: userDetails.referralCode,
+            bonusBalance: userDetails.bonusBalance,
+        });
+    }
+
+    public async findWithUserDetailsJoined(
```

Comment:
maybe we could just extend find method?

Reply:
I tried to do it, but it breaks existing logic of retrieving a current user.

Reply:
If it breaks the logic only in one place maybe it is worth updating it and having the one unified method?